### PR TITLE
fix(a11y): improve the second part of the donation modal

### PR DESCRIPTION
**Pull request by huyenltnguyen**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

This PR updates the donation modal to:
- Allow closing the modal by pressing Escape when the animation is complete
- Automatically focus on the donate button when the second part appears
  - There are two buttons on the modal: "Become a Supporter" and "Ask me later". The a11y recommendation is to have keyboard focus placed on a default focusable element in the modal. Generally the initial focus is set on the first focusable element, but we're free to decide what should receive the focus. I chose the "Become a Supporter" button because it's the first focusable element, but I'm open to suggestions here.
  - Ref: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/dialog_role#required_javascript_features

### CODE REVIEW

**Comment by bbsmooth** [comment]
MDN is a good accessibility reference, but it is not authoritative. In this case, I don't think MDN is giving you all of the options for managing focus in a modal dialog. You don't have to place the focus on an element in the dialog, you can place it on the dialog itself. Oh, actually, I just checked, and the [MDN page for the dialog element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog#accessibility) does actually suggest this.

I'm apparently not smart enough to trigger these modals so I can't give you my opinion on where focus should be placed. I just wanted to let you know that it doesn't have to be on either of these buttons.

**Comment by ahmaxed** [comment]
Thanks for the pr. I might need to AB test this to make sure it does not have any unintended side effects.
But the active state of the donation button came to my attention. It seems like we should either completely remove these or make them more subtle. 

I am leaning towards completey removing them similar to the paypal button. What do you think?

**Comment by ahmaxed** [comment]
The paypal should look similar to patreon as you mentioned. Some of that styles are coming through paypal's iframe.
Feel free to keep the changes introduced in this pr behind a11y-donation-modal flag. I will monitor it for a while to see if it affects donation at all.

File: client/src/components/Donation/donation-modal.tsx [hunk 1]
```diff
@@ -62,13 +66,22 @@ function DonateModal({
...
       goToAnchor('claim-cert-block');
     }
+
+    if (isA11yFeatureEnabled) {
+      if (!canClose) {
+        return;
+      }
+
+      closeDonationModal();
+    }
```
Comment (ahmaxed) [inline client/src/components/Donation/donation-modal.tsx:76]: Wouldn't the following be more straightforward. 
```suggestion
    if (isA11yFeatureEnabled && canClose ) {
              closeDonationModal();
    }
```

**Comment by ahmaxed** [comment]
Getting the following error for the variation with the new feature:

Screenshot 2024-08-16 at 1 55 04 PM

**Review by ahmaxed — APPROVED** [review]

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Pull request merged**